commerce-payment-method-mercanet/src/main/java/com/liferay/commerce/payment/method/mercanet/internal/MercanetCommercePaymentMethod.java

```diff
@@ -162,21 +161,19 @@ public CommercePaymentResult processPayment(
 		CommerceOrder commerceOrder = _commerceOrderService.getCommerceOrder(
 			mercanetCommercePaymentRequest.getCommerceOrderId());
 
-		CommerceCurrency commerceCurrency = commerceOrder.getCommerceCurrency();
+		if (!Objects.equals(
+				commerceOrder.getCommerceCurrencyCode(),
+				MercanetCommercePaymentMethodConstants.EUR_CURRENCY_CODE)) {
 
-		if (!Objects.equals(commerceCurrency.getCode(), "EUR")) {
 			throw new Exception("Mercanet accepts only EUR currency");
 		}
 
 		PaymentRequest paymentRequest = new PaymentRequest();
 
-		int normalizedMultiplier = (int)Math.pow(
-			10, commerceCurrency.getMaxFractionDigits());
-
 		BigDecimal orderTotal = commerceOrder.getTotal();
 
 		BigDecimal normalizedOrderTotal = orderTotal.multiply(
-			new BigDecimal(normalizedMultiplier));
+			new BigDecimal(100));
```

Comment:
Using `BigDecimal.valueOf` instead of `new BigDecimal` here could prevent needing to create a new object. Super minor performance consideration, but it could be worth making the change.